docs/models/advanced/itemoverridelist.md

```diff
@@ -0,0 +1,48 @@
+`ItemOverrideList`
+==================
+
+`ItemOverrideList` provides a way for an `IBakedModel` to process the state of an `ItemStack` and return a new `IBakedModel`; thereafter, the returned model replaces the old one. `ItemOverrideList` represents an arbitrary function `(IBakedModel, ItemStack, World, EntityLivingBase)` → `IBakedModel`, making it very useful for dynamic models. In vanilla, it is used to implement item property overrides.
+
+### `ItemOverrideList()`
+
+Given a list of `ItemOverride`s, the constructor copies that list and stores the copy. The list may be accessed with `getOverrides`, and it is used to implement the vanilla `applyOverride`, which, in turn, is used in the vanilla `handleItemState`.
+
+### <s>`applyOverride`</s>
+
+This is a deprecated vanilla method. It is only called in the vanilla `handleItemState`, and in almost all cases can be safely ignored.
+
+### `handleItemState`
+
+This simply takes an `IBakedModel`, an `ItemStack`, a `World`, and an `EntityLivingBase` to produce another `IBakedModel` to use for rendering. This method should not mutate the world in any way.
```

Comment:
mention that if modders want a custom item baked model this is where the interesting magic happens. handleItemState is called every time a baked model is rendered in an item context.